java-generator/core/src/main/java/io/fabric8/java/generator/nodes/JObject.java

```diff
@@ -280,6 +282,13 @@ public GeneratorResult generateJava() {
 
       objField.createGetter().addAnnotation("com.fasterxml.jackson.annotation.JsonAnyGetter");
       objField.createSetter().addAnnotation("com.fasterxml.jackson.annotation.JsonAnySetter");
+
+      MethodDeclaration additionalSetter = clz.addMethod("setAdditionalProperties", Modifier.Keyword.PUBLIC);
```

Comment:
```suggestion
      MethodDeclaration additionalSetter = clz.addMethod("setAdditionalProperty", Modifier.Keyword.PUBLIC);
```

Sorry, I may not be being clear.  I would expect both the test and this logic to say setAdditionalProperty singular.  That lines up with the model generation logic in the main project - Pod.setAdditionalProperty.

Reply:
Right, that's what I did, changing both breaks the functionality I'm fixing here as demonstrated by the additional test included in this PR.

If you are still in doubt I propose to have a quick sync on it to clarify somewhere next week ;-)

Reply:
Sorry, it would probably be best to compare to what's already in use.  The core model classes are generated with:

```
    @JsonIgnore
    private Map<String, Object> additionalProperties = new HashMap<String, Object>();

   @JsonAnyGetter
    public Map<String, Object> getAdditionalProperties() {
        return this.additionalProperties;
    }

    @JsonAnySetter
    public void setAdditionalProperty(String name, Object value) {
        this.additionalProperties.put(name, value);
    }
```

due to a limitation in the jsonschema2pojo, 

```
@JsonIgnore
void setAdditionalProperties(Map<String, Object> properties)
```

 being generated by the class annotations:

```
@Setter
@Accessors(prefix = {
    "_",
    ""
})
```

As far as I know that handles all of the issues we have seen with serdes and builder logic.

Could you try aligning to that?

Reply:
@shawkins sorry for the long forth and back, the integration test failure was due to a different reason ... fixed now 👍